src/main/flight/pid.c

```diff
@@ -213,16 +212,16 @@ static FAST_RAM_ZERO_INIT uint8_t itermRelaxCutoff;
 #endif
 
 #ifdef USE_RC_SMOOTHING_FILTER
-static FAST_RAM_ZERO_INIT pt1Filter_t setpointDerivativePt1[2];
-static FAST_RAM_ZERO_INIT biquadFilter_t setpointDerivativeBiquad[2];
+static FAST_RAM_ZERO_INIT pt1Filter_t setpointDerivativePt1[XYZ_AXIS_COUNT];
+static FAST_RAM_ZERO_INIT biquadFilter_t setpointDerivativeBiquad[XYZ_AXIS_COUNT];
 static FAST_RAM_ZERO_INIT bool setpointDerivativeLpfInitialized;
 static FAST_RAM_ZERO_INIT uint8_t rcSmoothingDebugAxis;
 static FAST_RAM_ZERO_INIT uint8_t rcSmoothingFilterType;
 #endif // USE_RC_SMOOTHING_FILTER
 
 void pidInitFilters(const pidProfile_t *pidProfile)
 {
-    BUILD_BUG_ON(FD_YAW != 2); // only setting up Dterm filters on roll and pitch axes, so ensure yaw axis is 2
+    BUILD_BUG_ON(FD_YAW != 2); // ensure yaw axis is 2
```

Comment:
This also seems redundant now

Reply:
I left it because it doesn't hurt anything and it would still be a real problem if for some reason roll/pitch/yaw got reordered (not sure why we would though).  There is still logic that expects to process roll/pitch by iterating through < FD_YAW.

Reply:
Haven't you removed all the cases of `< FD_YAW`? It was mostly there to avoid array overrun

Reply:
From an overflow perspective we should be fine, but there is still some logic that expects `FD_ROLL` and `FD_PITCH` to be less than `FD_YAW`.  Not just in the PID controller but elsewhere in the code as well.  So leaving this in place just helps catch any difficult to find problems if the ordering was ever changed.

Reply:
Also, we are heavily using `FD_YAW` as the stop element in loops over the axes (isn't that what Spitfire pilots did? :wink:), so making sure it's at the end is good.

It would probably be good if we changed these loops to use `< FD_AXIS_COUNT` as stop element instead at some point.